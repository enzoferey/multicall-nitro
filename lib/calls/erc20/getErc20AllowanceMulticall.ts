import type { Call } from "../../types";

export function getErc20AllowanceMulticall(
  erc20ContractAddress: string,
  ownerAddress: string,
  spenderAddress: string,
  options: { label?: string } = {}
): Call {
  const { label = "allowance" } = options;

  return {
    target: erc20ContractAddress,
    call: ["allowance(address,address)(uint256)", ownerAddress, spenderAddress],
    label,
  };
}
