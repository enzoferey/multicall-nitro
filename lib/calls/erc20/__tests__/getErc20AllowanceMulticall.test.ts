import { describe, it, expect } from "vitest";

import { getErc20AllowanceMulticall } from "../getErc20AllowanceMulticall";

describe("getErc20AllowanceMulticall", () => {
  it("should match the call", () => {
    expect(
      getErc20AllowanceMulticall(
        "0xdddddddddddddddddddddddddddddddddddddddd",
        "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
        "0xffffffffffffffffffffffffffffffffffffffff"
      )
    ).toMatchSnapshot();
  });
  it("should allow overwritting the value label", () => {
    expect(
      getErc20AllowanceMulticall(
        "0xdddddddddddddddddddddddddddddddddddddddd",
        "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
        "0xffffffffffffffffffffffffffffffffffffffff",
        { label: "some label" }
      ).label
    ).toMatchSnapshot();
  });
});
