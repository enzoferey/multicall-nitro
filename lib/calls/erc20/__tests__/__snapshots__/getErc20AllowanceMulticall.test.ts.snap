// Vitest Snapshot v1

exports[`getErc20AllowanceMulticall > should allow overwritting the value label 1`] = `"some label"`;

exports[`getErc20AllowanceMulticall > should match the call 1`] = `
{
  "call": [
    "allowance(address,address)(uint256)",
    "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
    "0xffffffffffffffffffffffffffffffffffffffff",
  ],
  "label": "allowance",
  "target": "0xdddddddddddddddddddddddddddddddddddddddd",
}
`;
